src/url-params.ts

```diff
@@ -0,0 +1,111 @@
+import * as _ from 'lodash';
+import {
+	getCookie,
+	setCookie,
+	hasCookieSupport,
+} from '@analytics/cookie-utils';
+import { Mixpanel } from 'mixpanel-browser';
+
+const URL_PARAM_DEVICE_ID = 'd_id';
+const COOKIES_DEVICE_IDS = '__analytics_dids';
+
+const deviceIdSeparator = /\s*,\s*/;
+
+const cookiesSupported = hasCookieSupport();
+
+/**
+ * AnalyticsUrlParams helps with handling analytics-related URL parameters
+ * like anonymous device identifier.
+ */
+export class AnalyticsUrlParams {
+	private deviceIds: string | null;
+
+	constructor(private mixpanel?: Mixpanel) {}
+
+	private storeInboundDeviceIds(ids: string, currentDeviceId: string | null) {
+		const list = ids ? ids.split(deviceIdSeparator) : [];
+		if (currentDeviceId) {
+			list.push(currentDeviceId);
+		}
+		const res = this.retrieveAndMerge(list);
+		this.deviceIds = res;
+		if (cookiesSupported) {
+			setCookie(COOKIES_DEVICE_IDS, res);
+		}
+		return res;
+	}
+
+	private retrieveAndMerge(input: string[]) {
```

Comment:
If you do what I suggested in the constructor, this won't be necessary (you can just use `this.deviceIds` directly). Then, the caller will turn it into a string if they wish to.

Reply:
This function is remove now.